core/templates/domain/topic/subtopic.model.ts

```diff
@@ -13,22 +13,21 @@
 // limitations under the License.
 
 /**
- * @fileoverview Factory for creating and mutating instances of frontend
+ * @fileoverview Model for creating and mutating instances of frontend
  * subtopic domain objects.
  */
 
-import { ShortSkillSummary } from
-  'domain/skill/short-skill-summary.model';
+import { ShortSkillSummary } from 'domain/skill/short-skill-summary.model';
 
 import constants from 'assets/constants';
 
 export interface SubtopicBackendDict {
   'id': number;
   'title': string;
   'skill_ids': string[];
-  'thumbnail_filename': string;
-  'thumbnail_bg_color': string;
-  'url_fragment': string;
+  'thumbnail_filename': string | null;
+  'thumbnail_bg_color': string | null;
+  'url_fragment': string | null;
```

Comment:
@seanlip Can I please keep the changes here, there are some reasons for it:
1. This file is dependency for other files covered here.
2. If I drop the changes here then there are some type errors.
3. This file is not covered fully here I'm covering this file separately - https://github.com/oppia/oppia/pull/13041 ,so as this PR gets in ,I can make the changes over there.
4. ` | null` is still needed just `createTitle` function has been removed over there.

Reply:
No, I'm afraid not; if there's a dependency, you'll need to make the other PR first, and only then merge this one, I think.

In general, we should never take the option of "merge a temporary bad fix into the codebase now, with the plan to fix it later". That just makes things messy.

Reply:
@seanlip  Done, dropped `read-only-topic-objecFactory` along with subtopic.model PTAL